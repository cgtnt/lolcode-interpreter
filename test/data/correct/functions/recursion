HAI

HOW IZ I ADD YR A AN YR B
  I HAS A RESULT ITZ SUM OF A AN B

  BOTH SAEM RESULT AN 5, O RLY?
    YA RLY
      FOUND YR I IZ ADD YR 1 AN YR 2 MKAY
    NO WAI
  OIC

  FOUND YR RESULT
IF U SAY SO

VISIBLE I IZ ADD YR 2 AN YR 3 MKAY!

KTHXBYE
