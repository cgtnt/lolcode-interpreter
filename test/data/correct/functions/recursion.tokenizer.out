@1:T_BEGIN:HAI-@3:T_COMMAND_TERMINATOR:
-@3:T_FUNC_BEGIN:HOW IZ I-@3:T_T_IDENTIFIER:ADD-@3:T_ARG:YR-@3:T_T_IDENTIFIER:A-@3:T_AND:AN-@3:T_ARG:YR-@3:T_T_IDENTIFIER:B-@4:T_COMMAND_TERMINATOR:
-@4:T_DECLARE_VAR:I HAS A-@4:T_T_IDENTIFIER:RESULT-@4:T_DECLARE_SET_VAR:ITZ-@4:T_PLUS:SUM OF-@4:T_T_IDENTIFIER:A-@4:T_AND:AN-@4:T_T_IDENTIFIER:B-@6:T_COMMAND_TERMINATOR:
-@6:T_EQUAL:BOTH SAEM-@6:T_T_IDENTIFIER:RESULT-@6:T_AND:AN-@6:T_T_INT:5-@6:T_COMMAND_TERMINATOR:,-@6:T_IF:O RLY?-@7:T_COMMAND_TERMINATOR:
-@7:T_THEN:YA RLY-@8:T_COMMAND_TERMINATOR:
-@8:T_RETURN_VAL:FOUND YR-@8:T_FUNC_CALL:I IZ-@8:T_T_IDENTIFIER:ADD-@8:T_ARG:YR-@8:T_T_INT:1-@8:T_AND:AN-@8:T_ARG:YR-@8:T_T_INT:2-@8:T_END_INF:MKAY-@9:T_COMMAND_TERMINATOR:
-@9:T_ELSE:NO WAI-@10:T_COMMAND_TERMINATOR:
-@10:T_END_IF:OIC-@12:T_COMMAND_TERMINATOR:
-@12:T_RETURN_VAL:FOUND YR-@12:T_T_IDENTIFIER:RESULT-@13:T_COMMAND_TERMINATOR:
-@13:T_FUNC_END:IF U SAY SO-@15:T_COMMAND_TERMINATOR:
-@15:T_WRITE_STDOUT:VISIBLE-@15:T_FUNC_CALL:I IZ-@15:T_T_IDENTIFIER:ADD-@15:T_ARG:YR-@15:T_T_INT:2-@15:T_AND:AN-@15:T_ARG:YR-@15:T_T_INT:3-@15:T_END_INF:MKAY-@15:T_BANG:!-@17:T_COMMAND_TERMINATOR:
-@17:T_END:KTHXBYE-@18:T_COMMAND_TERMINATOR:
-@18:T_EOF:
